zenohd_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.16)
+project(zenohd_vendor)
+
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+
+find_package(ament_cmake REQUIRED)
+
+include(ExternalProject)
+
+ExternalProject_Add(
+    zenoh
+    PREFIX ${CMAKE_BINARY_DIR}/_cargo_deps
+    GIT_REPOSITORY https://github.com/eclipse-zenoh/zenoh.git
+    GIT_TAG master
+    BUILD_COMMAND cargo build --package zenohd --release
+    CONFIGURE_COMMAND "" # Disable the configure step
+    INSTALL_COMMAND ""
+    BUILD_IN_SOURCE 1
+)
+
+# Install the zenohd executable from the rust crate
+install(
+    PROGRAMS ${CMAKE_BINARY_DIR}/_cargo_deps/src/zenoh/target/release/zenohd
+    DESTINATION bin
+)
+
+# TODO(francocipollone): Install zenoh router plugins
```

Comment:
I'm just trying this out, and when I try to run this zenohd I'm getting:

```
[2023-11-16T19:41:56Z INFO  zenohd] zenohd v0.11.0-dev-20-gd2399cad built with rustc 1.72.0 (5680fa18f 2023-08-23)
[2023-11-16T19:41:56Z INFO  zenohd] Initial conf: {"adminspace":{"permissions":{"read":true,"write":false}},"aggregation":{"publishers":[],"subscribers":[]},"connect":{"endpoints":[]},"id":"4d16dfcb3fce894934419fd3c1fa86da","listen":{"endpoints":["tcp/[::]:7447"]},"metadata":null,"mode":"router","plugins":{"rest":{"__required__":true,"http_port":"8000"}},"plugins_search_dirs":[],"queries_default_timeout":null,"routing":{"peer":{"mode":null},"router":{"peers_failover_brokering":null}},"scouting":{"delay":null,"gossip":{"autoconnect":null,"enabled":null,"multihop":null},"multicast":{"address":null,"autoconnect":null,"enabled":true,"interface":null,"listen":null},"timeout":null},"timestamping":{"drop_future_timestamp":null,"enabled":null},"transport":{"auth":{"pubkey":{"key_size":null,"known_keys_file":null,"private_key_file":null,"private_key_pem":null,"public_key_file":null,"public_key_pem":null},"usrpwd":{"dictionary_file":null,"password":null,"user":null}},"link":{"compression":{"enabled":false},"protocols":null,"rx":{"buffer_size":65535,"max_message_size":1073741824},"tls":{"client_auth":null,"client_certificate":null,"client_private_key":null,"root_ca_certificate":null,"server_certificate":null,"server_name_verification":null,"server_private_key":null},"tx":{"batch_size":65535,"keep_alive":4,"lease":10000,"queue":{"backoff":100,"size":{"background":1,"control":1,"data":4,"data_high":2,"data_low":2,"interactive_high":1,"interactive_low":1,"real_time":1}},"sequence_number_resolution":"32bit","threads":4},"unixpipe":{"file_access_mask":null}},"multicast":{"join_interval":2500,"max_sessions":1000},"qos":{"enabled":true},"shared_memory":{"enabled":false},"unicast":{"accept_pending":100,"accept_timeout":10000,"lowlatency":false,"max_links":1,"max_sessions":1000}}}
[2023-11-16T19:41:56Z INFO  zenohd] Loading required plugin "rest"
thread 'main' panicked at 'Plugin load failure: Library file 'libzenoh_plugin_rest.so' not found at commons/zenoh-util/src/std_only/lib_loader.rs:135.', zenohd/src/main.rs:97:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted (core dumped)
```

So I think we need to do this before we can merge this PR in.

Reply:
Mjm, wonder why it requested to load the rest API plugin when we haven't added to the config, I will double check.

Reply:
> I'm just trying this out, and when I try to run this zenohd I'm getting:
> 
> ```
> [2023-11-16T19:41:56Z INFO  zenohd] zenohd v0.11.0-dev-20-gd2399cad built with rustc 1.72.0 (5680fa18f 2023-08-23)
> [2023-11-16T19:41:56Z INFO  zenohd] Initial conf: {"adminspace":{"permissions":{"read":true,"write":false}},"aggregation":{"publishers":[],"subscribers":[]},"connect":{"endpoints":[]},"id":"4d16dfcb3fce894934419fd3c1fa86da","listen":{"endpoints":["tcp/[::]:7447"]},"metadata":null,"mode":"router","plugins":{"rest":{"__required__":true,"http_port":"8000"}},"plugins_search_dirs":[],"queries_default_timeout":null,"routing":{"peer":{"mode":null},"router":{"peers_failover_brokering":null}},"scouting":{"delay":null,"gossip":{"autoconnect":null,"enabled":null,"multihop":null},"multicast":{"address":null,"autoconnect":null,"enabled":true,"interface":null,"listen":null},"timeout":null},"timestamping":{"drop_future_timestamp":null,"enabled":null},"transport":{"auth":{"pubkey":{"key_size":null,"known_keys_file":null,"private_key_file":null,"private_key_pem":null,"public_key_file":null,"public_key_pem":null},"usrpwd":{"dictionary_file":null,"password":null,"user":null}},"link":{"compression":{"enabled":false},"protocols":null,"rx":{"buffer_size":65535,"max_message_size":1073741824},"tls":{"client_auth":null,"client_certificate":null,"client_private_key":null,"root_ca_certificate":null,"server_certificate":null,"server_name_verification":null,"server_private_key":null},"tx":{"batch_size":65535,"keep_alive":4,"lease":10000,"queue":{"backoff":100,"size":{"background":1,"control":1,"data":4,"data_high":2,"data_low":2,"interactive_high":1,"interactive_low":1,"real_time":1}},"sequence_number_resolution":"32bit","threads":4},"unixpipe":{"file_access_mask":null}},"multicast":{"join_interval":2500,"max_sessions":1000},"qos":{"enabled":true},"shared_memory":{"enabled":false},"unicast":{"accept_pending":100,"accept_timeout":10000,"lowlatency":false,"max_links":1,"max_sessions":1000}}}
> [2023-11-16T19:41:56Z INFO  zenohd] Loading required plugin "rest"
> thread 'main' panicked at 'Plugin load failure: Library file 'libzenoh_plugin_rest.so' not found at commons/zenoh-util/src/std_only/lib_loader.rs:135.', zenohd/src/main.rs:97:21
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> Aborted (core dumped)
> ```
> 
> So I think we need to do this before we can merge this PR in.

Oh you run it via`zenohd` command, right? 

Note that in your log message, the `rest` plugin is part of the configuration file but if you check our config file, you won't find it.

`zenohd` by default uses a configuration in which it uses rest API plugin. In our case we are not using plugins and we don't need to bring the plugins for the moment. 
If you want to test it: (Sorry I haven't added a `test it` section in the PR description) you have two options:
 1. Run ` zenohd -c <path to our config file>` : Use the binary we brought but make sure to pass a configuration file without plugin use.
 2. Run `ros2_run rmw_zenoh_cpp init_rmw_zenoh_router`: Here internally we are using our configuration file.

Reply:
> Oh you run it via`zenohd` command, right?

Oh, right.  That was my mistake.  Running it with `ros2 run` made it work for me.  Sorry about that.